tests/main/prepare-image-reproducible/task.yaml

```diff
@@ -0,0 +1,138 @@
+summary: Verify that we are able to request specific revisions of snaps from the store through prepare-image 
+
+# autopkgtest run only a subset of tests that deals with the integration
+# with the distro
+backends: [-autopkgtest]
+
+environment:
+    ROOT: /home/test/tmp/
+    IMAGE: /home/test/tmp/image
+    GADGET: /home/test/tmp/gadget
+    STORE_DIR: $(pwd)/fake-store-blobdir
+    STORE_ADDR: localhost:11028
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    mkdir -p "$ROOT"
+    chown test:test "$ROOT"
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    teardown_fake_store "$STORE_DIR"
+    
+    rm -rf "$ROOT"
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    function install_snap_to_fakestore() {
```

Comment:
(nipitck) we probably should to remove `function` here, all our prepare/restore scripts do not use so it's more consistent this way (and it seems like it's not standardized but we use bash so that's fine). So just  `install_snap_to_fakestore() {` should be fine :)

Reply:
Ah cool, actually never thought about this, thanks for pointing this out!